client/signup/config/flows.js

```diff
@@ -228,7 +229,8 @@ function removeUserStepFromFlow( flow ) {
 }
 
 function filterFlowName( flowName ) {
-	if ( 'headstart' === flowName && 'production' === config( 'env' ) ) {
+	const locale = getLocaleSlug();
+	if ( 'headstart' === flowName && 'en' !== locale && 'en-gb' !== locale ) {
 		return 'main';
```

Comment:
No – that's the check that says, "if we're in the Headstart flow, but the user's locale isn't EN, send them to `main` instead" (which is localized).